indico/modules/events/abstracts/controllers/reviewing.py

```diff
@@ -51,8 +58,96 @@ class RHAbstractsDownloadAttachment(RHAbstractsReviewBase):
+                 .filter(Abstract.id.notin_(self.excluded_ids))
+                 .order_by(Abstract.friendly_id))
+
+        result = [{'id': abstract.id, 'friendly_id': abstract.friendly_id, 'title': abstract.title,
+                   'full_title': '#{}: {}'.format(abstract.friendly_id, abstract.title)}
+                  for abstract in query
+                  if abstract.can_access(session.user)]
+        return jsonify(result)
+
+
+class RHJudgeAbstract(RHAbstractReviewBase):
+    CSRF_ENABLED = True
+
+    def _process(self):
+        form = AbstractJudgmentForm(abstract=self.abstract)
+        if form.validate_on_submit():
+            judgment_data, abstract_data = form.split_data
+            judge_abstract(self.abstract, abstract_data, judge=session.user, **judgment_data)
+            return jsonify_data(page_html=render_abstract_page(self.abstract, management=self.management))
+
+        tpl = get_template_module('events/abstracts/abstract/judge.html')
+        return jsonify_data(box_html=tpl.render_decision_box(self.abstract, form, management=self.management))
+
+
+class RHResetAbstractJudgment(RHAbstractReviewBase):
+    def _process(self):
+        if self.abstract.state not in (AbstractState.submitted, AbstractState.withdrawn):
+            reset_abstract_judgment(self.abstract)
+            flash(_("Abstract judgment has been reset"), 'success')
+        html = render_abstract_page(self.abstract, management=self.management)
+        return jsonify_data(display_html=html, management_html=html)
+
+
+class RHReviewAbstractForTrack(RHAbstractReviewBase):
+    CSRF_ENABLED = True
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.abstract
+        },
+        'preserved_args': {'track_id'}
+    }
+
+    def _checkProtection(self):
+        if not self.abstract.can_review(session.user, check_state=True):
+            raise Forbidden
+        RHAbstractReviewBase._checkProtection(self)
+
+    def _checkParams(self, params):
+        RHAbstractReviewBase._checkParams(self, params)
+        self.track = Track.get_one(request.view_args['track_id'])
+        reviews = self.abstract.get_reviews(user=session.user, track=self.track)
+        self.review = reviews[0] if reviews else None
+
+    def _process(self):
+        form = build_review_form(self.abstract, self.track)
+
+        if form.validate_on_submit():
+            if self.review:
+                form.populate_obj(self.review)
+                for question in self.event_new.abstract_review_questions:
+                    rating = question.get_review_rating(self.review)
+                    if not rating:
+                        rating = AbstractReviewRating(question=question, review=self.review)
+                    rating.value = int(form.data["question_{}".format(question.id)])
+            else:
+                self.review = AbstractReview(abstract=self.abstract, track=self.track, user=session.user)
```

Comment:
I would move all of this into a `review_abstract` function in `operations.py` and add some logging.